chatkit-core/src/main/kotlin/com/pusher/chatkit/PresenceSubscription.kt

```diff
@@ -1,74 +1,63 @@
 package com.pusher.chatkit
 
-import com.pusher.chatkit.User.Presence.Offline
-import com.pusher.chatkit.User.Presence.Online
+import com.pusher.chatkit.ChatManagerEvent.*
 import com.pusher.chatkit.network.parseAs
 import com.pusher.platform.Instance
 import com.pusher.platform.SubscriptionListeners
+import com.pusher.platform.network.map
+import com.pusher.platform.network.toFuture
+import com.pusher.platform.network.wait
 import com.pusher.platform.tokenProvider.TokenProvider
 import com.pusher.util.asFailure
-import com.pusher.util.fold
+import com.pusher.util.mapResult
 import elements.Errors
 import elements.Subscription
-import kotlinx.coroutines.experimental.channels.BroadcastChannel
-import kotlinx.coroutines.experimental.channels.Channel
+import java.util.concurrent.Future
 
 class PresenceSubscription(
     val instance: Instance,
     path: String,
     val tokenProvider: TokenProvider,
     val tokenParams: ChatkitTokenParams?,
     private val chatManager: ChatManager,
-    private val events: BroadcastChannel<ChatManagerEvent> = BroadcastChannel(Channel.CONFLATED)
-) : BroadcastChannel<ChatManagerEvent> by events {
+    private val consumeEvent: (ChatManagerEvent) -> Unit
+) : Subscription {
 
-    var subscription: Subscription
+    private val subscription: Subscription = instance.subscribeResuming<ChatEvent>(
+        path = path,
+        tokenParams = tokenParams,
+        tokenProvider = tokenProvider,
+        listeners = SubscriptionListeners(
+            onEvent = { event ->
+                event.body
+                    .toUserPreferences()
```

Comment:
`UserPresences` maybe?

Reply:
???

Reply:
Ha. Fair.

I meant:

what is `UserPreferences`?

Did you mean to write `UserPresences`?

Reply:
ah! yeah good spot :D